File: dev-docs/snippets/errors/match-code.ts
import type { JsBaoClient } from "js-bao-wss-client";
import { isJsBaoError } from "js-bao-wss-client";

// Match on a specific error code. `code` is the string-literal union
// `JsBaoErrorCode`, so compare it directly against a wire string.
export async function matchCode(client: JsBaoClient, userId: string) {
  // #region example
  try {
    await client.users.getBasic(userId);
  } catch (e) {
    if (isJsBaoError(e) && e.code === "NOT_FOUND") {
      console.warn("no such user; showing empty state");
      return;
    }
    throw e;
  }
  // #endregion example
}

File: dev-docs/snippets/errors/match-code.swift
import JsBaoClient

// Match on a specific error code. `code` is a `JsBaoErrorCode` enum, so
// compare against a case (here `.notFound`) rather than a raw string.
func matchCode(client: JsBaoClient, userId: String) async throws {
  // #region example
  do {
    _ = try await client.users.getBasic(userId: userId)
  } catch let err as JsBaoError where err.code == .notFound {
    print("no such user; showing empty state")
  }
  // #endregion example
}
